packages/deploy-utils/src/build-and-push/index.ts

```diff
@@ -8,7 +8,8 @@ import { pushToS3 } from "../utils/s3";
  * @param bucket - bucket to push to
  */
 export const buildAndPush = (bucket: string) => {
-  execSync("yarn build", { stdio: "inherit" });
+  // Disable script mode to let vite build the project
+  execSync("VITE_SCRIPT_MODE=0 yarn build", { stdio: "inherit" });
```

Comment:
Alternatively, you could add the blocking code specifically on `serverConfig` i.e.

```
const serverConfig = useProjectConfig ? { // real config // } : {}
```

and you wouldn't need the environment variable

```suggestion
  execSync("yarn build", { stdio: "inherit" });
```